swap/swap.go

```diff
@@ -1,98 +1,451 @@
+type Owner struct {
+	Contract   common.Address    // address of swap contract
+	address    common.Address    // owner address
+	privateKey *ecdsa.PrivateKey // private key
+	publicKey  *ecdsa.PublicKey  // public key
+}
+
+// Params encapsulates param
+type Params struct {
+	InitialDepositAmount uint64 //
+}
+
+// NewParams returns a Params struct filled with default values
+func NewParams() *Params {
+	return &Params{
+		InitialDepositAmount: DefaultInitialDepositAmount,
+	}
 }
 
 // New - swap constructor
-func New(stateStore state.Store) (swap *Swap) {
-	swap = &Swap{
-		stateStore: stateStore,
-		balances:   make(map[enode.ID]int64),
+func New(stateStore state.Store, prvkey *ecdsa.PrivateKey, contract common.Address, backend contract.Backend) *Swap {
+	sw := &Swap{
+		store:               stateStore,
+		balances:            make(map[enode.ID]int64),
+		backend:             backend,
+		cheques:             make(map[enode.ID]*Cheque),
+		peers:               make(map[enode.ID]*Peer),
+		params:              NewParams(),
+		paymentThreshold:    DefaultPaymentThreshold,
+		disconnectThreshold: DefaultDisconnectThreshold,
+		oracle:              NewPriceOracle(),
 	}
-	return
+	sw.owner = sw.createOwner(prvkey, contract)
+	return sw
+}
+
+const (
+	balancePrefix        = "balance_"
+	sentChequePrefix     = "sent_cheque_"
+	receivedChequePrefix = "received_cheque_"
+)
+
+// returns the store key for retrieving a peer's balance
+func balanceKey(peer enode.ID) string {
+	return balancePrefix + peer.String()
+}
+
+// returns the store key for retrieving a peer's last sent cheque
+func sentChequeKey(peer enode.ID) string {
+	return sentChequePrefix + peer.String()
+}
+
+// returns the store key for retrieving a peer's last received cheque
+func receivedChequeKey(peer enode.ID) string {
+	return receivedChequePrefix + peer.String()
+}
+
+func keyToID(key string, prefix string) enode.ID {
+	return enode.HexID(key[len(prefix):])
+}
+
+// createOwner assings keys and addresses
+func (s *Swap) createOwner(prvkey *ecdsa.PrivateKey, contract common.Address) *Owner {
+	pubkey := &prvkey.PublicKey
+	return &Owner{
+		privateKey: prvkey,
+		publicKey:  pubkey,
+		Contract:   contract,
+		address:    crypto.PubkeyToAddress(*pubkey),
+	}
+}
+
+// DeploySuccess is for convenience log output
+func (s *Swap) DeploySuccess() string {
+	return fmt.Sprintf("contract: %s, owner: %s, deposit: %v, signer: %x", s.owner.Contract.Hex(), s.owner.address.Hex(), s.params.InitialDepositAmount, s.owner.publicKey)
 }
 
-//Swap implements the protocols.Balance interface
-//Add is the (sole) accounting function
+// Add is the (sole) accounting function
+// Swap implements the protocols.Balance interface
 func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	//load existing balances from the state store
-	err = s.loadState(peer)
+	// load existing balances from the state store
```

Comment:
please reduce the amounts of comments in this function. there's more comments than code here. the comment on this specific line can be removed, also on 159, 163, 174. I'm all in for comments, but please try to condense them

Reply:
comments condensed and removed.